File: src/cs/PhantasyIslandPythonRemoteControl/pp/radio/api_fly.py
"""
飞行器相关 API 模块。
"""

from __future__ import annotations

from .api_module import ApiModule, SendResult


class FlyApi(ApiModule):

    def listFlyObject(self) -> SendResult[dict]:
        return self.send('fly.listFlyObject')

    def getFlyObjectInfo(self, keyName: str) -> SendResult[dict]:
        return self.send('fly.getFlyObjectInfo', data={'keyName': keyName})

    def getFlyObjectCameraImageDown(self, keyName: str) -> SendResult[dict]:
        return self.send('fly.getFlyObjectCameraImageDown', data={'keyName': keyName})

    def getFlyObjectCameraImageFront(self, keyName: str) -> SendResult[dict]:
        return self.send('fly.getFlyObjectCameraImageFront', data={'keyName': keyName})

File: src/el/radio/api_fly.erl
-module(api_fly).
-export([listFlyObject/1, getFlyObjectInfo/2, getFlyObjectCameraImageDown/2, getFlyObjectCameraImageFront/2]).

listFlyObject(State) ->
    api_module:send(State, "fly.listFlyObject", #{}).

getFlyObjectInfo(State, KeyName) ->
    api_module:send(State, "fly.getFlyObjectInfo", #{<<"keyName">> => KeyName}).

getFlyObjectCameraImageDown(State, KeyName) ->
    api_module:send(State, "fly.getFlyObjectCameraImageDown", #{<<"keyName">> => KeyName}).

getFlyObjectCameraImageFront(State, KeyName) ->
    api_module:send(State, "fly.getFlyObjectCameraImageFront", #{<<"keyName">> => KeyName}).
